File: backend/Models/POI.cs

namespace GeoInformation.Models
{
    public record POI
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public required string Description { get; init; }
        public required string Category { get; init; }
        public required double Latitude { get; init; }
        public required double Longitude { get; init; }
        public required DateTime CreationTime { get; init; }
    }
}

File: pois-frontend/src/Models/POI.ts
import * as z from 'zod'

export const POISchema = z.object({
    id: z.string(),
    category: z.string(),
    name: z.string(),
    description: z.string(),
    latitude: z.number(),
    longitude: z.number()
})

export type POI = z.infer<typeof POISchema>;